app/models/miq_set.rb

```diff
@@ -0,0 +1,66 @@
+class MiqSet < ApplicationRecord
+  # self.abstract_class = true
+  self.inheritance_column = :set_type
+
+  include UuidMixin
+
+  acts_as_miq_taggable
+
+  serialize :set_data
+
+  validates :name, :presence => true, :uniqueness_when_changed => {:scope => [:set_type, :userid, :group_id], :if => proc { |c| c.class.in_my_region.exists?(:name => c.name) }}
+  validates :description, :presence => true
+
+  has_many :miq_set_memberships, :dependent => :delete_all
+  belongs_to :owner, :polymorphic => true
+
+  attr_writer :model_class
+
+  def self.model_class
+    @model_class ||= name[0..-4].constantize
```

Comment:
I'd prefer this were `name.chomp("Set").constantize` to more clearly illustrate what this code is doing.